src/core/thread/mlr_manager.cpp

```diff
@@ -289,9 +399,25 @@ void MlrManager::SendMulticastListenerRegistration(void)
     SuccessOrExit(message->SetPayloadMarker());
 
     addressesTlv.Init();
-    addressesTlv.SetLength(sizeof(Ip6::Address) * addressesNum);
+    addressesTlv.SetLength(sizeof(Ip6::Address) * aAddressNum);
     SuccessOrExit(error = message->Append(&addressesTlv, sizeof(addressesTlv)));
-    SuccessOrExit(error = message->Append(&addresses, sizeof(Ip6::Address) * addressesNum));
+    SuccessOrExit(error = message->Append(aAddresses, sizeof(Ip6::Address) * aAddressNum));
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE && OPENTHREAD_CONFIG_COMMISSIONER_ENABLE
+    if (Get<MeshCoP::Commissioner>().IsActive())
+    {
+        SuccessOrExit(error = ThreadTlv::AppendUint16Tlv(*message, ThreadTlv::kCommissionerSessionId,
+                                                         Get<MeshCoP::Commissioner>().GetSessionId()));
+    }
+
+    if (aTimeout != nullptr)
+    {
+        SuccessOrExit(error = Tlv::AppendUint32Tlv(*message, ThreadTlv::kTimeout, *aTimeout));
+    }
+#else
+    OT_UNUSED_VARIABLE(aTimeout);
```

Comment:
nit: move to the beginning.

https://github.com/openthread/openthread/blob/master/STYLE_GUIDE.md

> Unused parameters
All unused parameters shall be declared as such using the OT_UNUSED_VARIABLE macro at the top of a function or method before all local variable declarations.